src/com/gardev/neuralnetwork/NeuralNetwork.java

```diff
@@ -47,6 +55,52 @@ public int test(double... inputs) {
         return output.getResult();
     }
 
+    public double[] getOutputs(double... inputs) {
+        activate(inputs);
+        return output.getOutputs();
+    }
+
+    public void serializeNetwork(OutputStream stream) {
+        if (stream != null) {
+            try {
+                ZipOutputStream zipout = new ZipOutputStream(stream);
+                zipout.setLevel(9);
+                zipout.putNextEntry(new ZipEntry("data"));
+                ObjectOutputStream out = new ObjectOutputStream(zipout);
+
+                input.serialize(out);
+                for(int i = 0; i < hidden.size(); i++) {
+                    hidden.get(i).serialize(out);
+                }
+
+                out.flush();
+                zipout.closeEntry();
+                out.close();
+            } catch (Exception E) {
```

Comment:
The typical convention is to have variable names in camelCase. The correct usage will be `catch (Exception e)`